src/lib.rs

```diff
@@ -0,0 +1,141 @@
+extern crate diesel;
+extern crate byteorder;
+extern crate geo;
+
+use byteorder::{ByteOrder, LittleEndian, BigEndian};
+use std::error::Error;
+use std::io::prelude::*;
+use diesel::pg::{Pg, PgTypeMetadata};
+use diesel::query_builder::QueryId;
+use diesel::row;
+use diesel::types::{FromSql, ToSql, IsNull, HasSqlType, FromSqlRow};
+use std::fmt;
+use geo::*;
+
+
+pub struct Geography {
+    pub big_endian: bool,
+    pub srid: Option<u32>,
+    pub geom: geo::Geometry<f64>,
+}
+
+#[derive(Debug)]
+struct NotImplemented {}
+impl fmt::Display for NotImplemented {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Not implemented Error")
+    }
+}
+
+impl Error for NotImplemented {
+    fn description(&self) -> &str {
+        "Not implemented"
+    }
+}
+
+impl Geography {
+    pub fn from(bytes: &[u8]) -> Result<Self, Box<Error + Send + Sync>> {
```

Comment:
The tiniest of nits in comparison to the other feedback: You might want to rename this method to `try_from` as it looks like the one from the `std::convert::TryFrom` trait, which is sadly not yet stable as of Rust 1.12 (cf. <https://github.com/rust-lang/rust/issues/33417>).